content/tutorial/tutorial.md

```diff
@@ -1170,41 +1177,41 @@ We will also replace reading `this.state.history` with `this.state.history.slice
   }
 ```
 
-Finally, we will modify the Game component's `render` method from always rendering the last move to rendering the currently selected move according to `stepNumber`:
+अंत में, हम Game कौम्पोनॅन्ट के `render` मेथड को हमेशा अंतिम चरण प्रदान करने से लेकर वर्तमान में चुने गए कदम को `stepNumber` के अनुसार रेंडर करेंगे।
 
 ```javascript{3}
   render() {
     const history = this.state.history;
     const current = history[this.state.stepNumber];
     const winner = calculateWinner(current.squares);
 
-    // the rest has not changed
+    // और इसके आगे कोई चाल बाकि नहीं है
 ```
 
-If we click on any step in the game's history, the tic-tac-toe board should immediately update to show what the board looked like after that step occurred.
+यदि हम गेम के history के किसी भी चरण पर क्लिक करते हैं, तो टिक-टैक-टो Board को तुरंत यह दिखाने के लिए अपडेट करना चाहिए कि Board उस चरण के बाद कैसा दिखता था।
 
-**[View the full code at this point](https://codepen.io/gaearon/pen/gWWZgR?editors=0010)**
+**[अभी तक का पूरा कोड देखें](https://codepen.io/gaearon/pen/gWWZgR?editors=0010)**
 
-### Wrapping Up {#wrapping-up}
+### अंतिम स्टेप्स {#wrapping-up}
 
-Congratulations! You've created a tic-tac-toe game that:
+बधाई हो! आपने एक टिक-टैक-टो गेम बनाया है जो:
 
-* Lets you play tic-tac-toe,
-* Indicates when a player has won the game,
-* Stores a game's history as a game progresses,
-* Allows players to review a game's history and see previous versions of a game's board.
+* अब टिक-टैक-टो खेलते हैं,
+* संकेत करता है कि जब किसी खिलाड़ी ने गेम जीता है,
+* एक खेल के रूप में एक खेल के इतिहास को स्टोर करता है,
+* खिलाड़ियों को गेम के इतिहास की समीक्षा करने और गेम के बोर्ड के पिछले संस्करणों को देखने की अनुमति देता है।
 
-Nice work! We hope you now feel like you have a decent grasp on how React works.
+अच्छा काम! हमें उम्मीद है कि अब आप महसूस करेंगे कि आपके पास React कैसे काम करता है, इस पर एक सभ्य समझ है।
 
-Check out the final result here: **[Final Result](https://codepen.io/gaearon/pen/gWWZgR?editors=0010)**.
+अंतिम परिणाम यहां देखें: **[अंतिम परिणाम](https://codepen.io/gaearon/pen/gWWZgR?editors=0010)**.
 
-If you have extra time or want to practice your new React skills, here are some ideas for improvements that you could make to the tic-tac-toe game which are listed in order of increasing difficulty:
+यदि आपके पास अतिरिक्त समय है या आप अपने नए React कौशल का अभ्यास करना चाहते हैं, तो यहां सुधार के लिए कुछ विचार दिए गए हैं जो आप टिक-टैक-टो खेल में कर सकते हैं जो कि बढ़ती कठिनाई के क्रम में सूचीबद्ध हैं:
```

Comment:
यदि आपके पास अतिरिक्त समय है  incorrect translation

Reply:
सूचीबद्ध could be removed altogether?

Reply:
> यदि आपके पास अतिरिक्त समय है incorrect translation

`यदि आपके पास और समय बाकि है या आप अपने नए React स्किल्स का अभ्यास करना चाहते हैं` ?

Reply:
sounds good